src/value.jl

```diff
@@ -146,8 +146,10 @@ Base.eltype(x::CategoricalString) = Char
 Base.length(x::CategoricalString) = length(get(x))
 Base.endof(x::CategoricalString) = endof(get(x))
 Base.sizeof(x::CategoricalString) = sizeof(get(x))
-Base.nextind(x::CategoricalString, i::Integer) = nextind(get(x), i)
-Base.prevind(x::CategoricalString, i::Integer) = prevind(get(x), i)
+Base.nextind(x::CategoricalString, i::Integer) = nextind(x, Int(i))
+Base.nextind(x::CategoricalString, i::Int) = nextind(get(x), i)
```

Comment:
No need to define a method for `::Int`, the conversion used in the `::Integer` method will be a no-op anyway.